tutorials/notebooks/synphot-model-spec/skymodel.py

```diff
@@ -0,0 +1,202 @@
+        (default is 300.0)
+    wmax: float
+        Maximum wavelength (nm) in the wavelength grid.
+        Must be in range [300.0,30000.0] and > wmin
+        (default is 2000.0)
+    wgrid_mode: str
+        Mode of the wavelength grid.
+        options: ['fixed_spectral_resolution','fixed_wavelength_step', 'user']
+        (default is 'fixed_wavelength_step')
+    wdelta: float
+        Wavelength sampling step dlam in range [0,30000.0] (nm/step)
+        (default is 0.1)
+    wres: int
+        lam/dlam where dlam is wavelength step.
+        In range [0,1.0e6] (default is 20000)
+    wgrid_user: list of floats
+        default is [500.0, 510.0, 520.0, 530.0, 540.0, 550.0]
+
+    lsf_type: str
+        Line spread function type for convolution.
+        options: ['none','Gaussian','Boxcar'] (default is 'none')
+    lsf_gauss_fwhm: float
+        Gaussian full-width half-max for line spread function wavelength bins.
+        Range > 0.0 (default is 5.0)
+    lsf_boxcar_fwhm: float
+        Boxcar full-width half-max for line spread function wavelength bins.
+        Range > 0.0 (default is 5.0)
+
+    observatory: str
+        Observatory where observation takes place.
+        Options are 'paranal', 'lasilla', 'armazones' (default is 'paranal')
+
+    Returns
+    -------
+    trans_waves, transmission: tuple of arrays of floats
+        'trans_waves' is an array of wavelengths in angstroms (float),
+        'transmission' is an array of fractional atmospheric
+        transmittance (float).
+
+    """
+
+    params = locals()
+
+    if params['observatory'] == 'lasilla':
+        params['observatory'] = '2400'
+    elif params['observatory'] == 'paranal':
+        params['observatory'] = '2640'
+    elif (params['observatory'] == '3060m' or
+          params['observatory'] == 'armazones'):
+        params['observatory'] = '3060'
+    else:
+        raise ValueError('Wrong Observatory name, please refer to the '
+                         'skycalc_cli documentation.')
+
+    # Use the bit from skycalc_cli which queries from the SkyCalc Sky Model
+    server = 'http://etimecalret-001.eso.org'
+    url = server + '/observing/etc/api/skycalc'
+    response = requests.post(url, data=json.dumps(params))
+    results = json.loads(response.text)
+
+    status = results['status']
+    tmpdir = results['tmpdir']
+    tmpurl = server + '/observing/etc/tmp/' + tmpdir + '/skytable.fits'
+
+    if status == 'success':
+        try:
+            response = requests.get(tmpurl, stream=True)
+            data = response.content
+        except requests.exceptions.RequestException as e:
+            print(e, 'could not retrieve FITS data from server')
+    else:
+        print('HTML request failed', results)
+
+    # Create a temporary file to write the binary results to
+    tmp_data_file = './tmp_skycalc_data.fits'
+
+    with open(tmp_data_file, 'wb') as f:
+        f.write(data)
+
+    hdu = fits.open(tmp_data_file)
```

Comment:
I'm having trouble using fits for saving, since `data` is a bytearray. For example, I try to replace lines 192 and 193 with
```
openhdu = fits.BinTableHDU(Table(data, dtype='bytearray'))
    openhdu.writeto(tmp_data_file)
```
and get the value error `ValueError: Data type <class 'bytes'> not allowed to init Table`. There's probably a better way to go about this that I'm not seeing... otherwise I could just add `f.close()`?

(long time no see, btw! Hope you're doing relatively well in this brave new world)

Reply:
👋 @tcjansen Where are you getting that error?

Reply:
Yeah, will need more info to reproduce the error on our side...

Reply:
@adrn @kakirastern This happens in part 5 of the synphot-model-spec notebook, specifically in the call to my local version of `skymodel.py`, where I'm testing using `astropy.io.fits` to write the data received from a web request to a fits file and to close the handler on exit, as @pllim suggested (the way I have it in the current commit left the file open). I think the solution I proposed above would work fine if the requested `data` weren't in bytes, since it appears `astropy.table.Table` won't work with bytes. Alternatively, I could just manually close the file with `f.close()` at line 194 in `skymodel`, unless there are objections to that.
I hope that clears up some confusion... let me know if I can provide more info! Here's the full traceback from my local version of `skymodel.py` (which is only different by the two lines in question):

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-11-c6c183f1d026> in <module>
----> 1 trans_waves, transmission = skymodel.get_atmospheric_transmittance(airmass=1.5)
      2 
      3 atmosphere = SpectralElement(Empirical1D,
      4                              points=trans_waves,
      5                              lookup_table=transmission)

/Volumes/external/github/astropy-tutorials/skymodel.py in get_atmospheric_transmittance(airmass, pwv_mode, season, time, pwv, msolflux, incl_moon, moon_sun_sep, moon_target_sep, moon_alt, moon_earth_dist, incl_starlight, incl_zodiacal, ecl_lon, ecl_lat, incl_loweratm, incl_upperatm, incl_airglow, incl_therm, therm_t1, therm_e1, therm_t2, therm_e2, therm_t3, therm_e3, vacair, wmin, wmax, wgrid_mode, wdelta, wres, lsf_type, lsf_gauss_fwhm, lsf_boxcar_fwhm, observatory, keep_tmp_data_file)
    201     # with open(tmp_data_file, 'wb') as f:
    202     #     f.write(data)
--> 203     openhdu = fits.BinTableHDU(Table(data, dtype='bytearray'))
    204     openhdu.writeto(tmp_data_file)
    205 

~/anaconda3/lib/python3.7/site-packages/astropy/table/table.py in __init__(self, data, masked, names, dtype, meta, copy, rows, copy_indices, **kwargs)
    514         else:
    515             raise ValueError('Data type {0} not allowed to init Table'
--> 516                              .format(type(data)))
    517 
    518         # Set up defaults if names and/or dtype are not specified.

ValueError: Data type <class 'bytes'> not allowed to init Table
```

Reply:
I don't remember much from GSoC last year anymore. Did this ever work before? What if you do `data = response.text` and avoid bytes?